java/ql/src/experimental/Security/CWE/CWE-089/MyBatisCommonLib.qll

```diff
@@ -152,6 +153,8 @@ predicate isMybatisXmlOrAnnotationSqlInjection(
         unsafeExpression.matches("${param" + (i + 1) + "%}")
         or
         unsafeExpression.matches("${arg" + i + "%}")
+        or
+        unsafeExpression.matches("${" + ma.getMethod().getParameter(i).getName() + "}")
```

Comment:
```suggestion
        unsafeExpression.matches("${" + ma.getMethod().getParameter(i).getName() + "%}")
```
The `%` operator matches any string, and in this case could consume a modifier like `,jdbcType=VARCHAR` as seen in the example.

I note a small bug in this and the existing cases btw, when a method parameter's name is a prefix of another one, or the same situation for fields. If you had the appetite for it, a good future PR would be to switch to regex matching, and check that the suffix begins with optional whitespace followed by a comma.

Reply:
I notice the lack of '%'. My first feeling is "Yes, add it. All of others added it.".
But after test, I noticed Mybatis not support usage like "${var,jdbcType=VARCHAR}". This makes an error occur.
`Parameter 'VARCHAR' not found. Available parameters are [var, param1]`

Reply:
But you can use `${param1,jdbcType=VARCHAR}`?

Reply:
Interesting. It's a little concerning considering the `%` on the param case clearly expects that you can... but at the least this query is better than it was, so I'm happy to merge it.

Reply:
> But you can use `${param1,jdbcType=VARCHAR}`?

I can use `#{  param1  ,  jdbcType=VARCHAR  }` but not `${param1,jdbcType=VARCHAR}`.
Another small bug , see the space :(

Reply:
> Interesting. It's a little concerning considering the `%` on the param case clearly expects that you can... but at the least this query is better than it was, so I'm happy to merge it.

I'm wrong, dude.
On annotation SQL statement, `'${param1,jdbcType=VARCHAR}'` is failed. I do not know the reason.
On XML SQL statement, `'${param1,jdbcType=VARCHAR}'` is working correctly.